### PR TITLE
[IMP] web: improve scss to hide field in edit mode of list view

### PR DESCRIPTION
**Pull request by jip-odoo**
Task: https://www.odoo.com/web?#id=1949618&action=327&model=project.task&view_type=form&menu_id=4720
Pad: https://pad.odoo.com/p/r.a80bef5d908739485b8ca61ee4595be8

task- 1949618

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by msh-odoo** [comment]
LGTM, @aab-odoo would you please check this, @jip-odoo need to improve commit message

File: addons/web/static/src/scss/form_view.scss [hunk 1]
```diff
@@ -742,6 +742,9 @@
     .o_field_widget .o_list_view {
         margin-bottom: 10px;
 
+        > thead > tr > th.oe_read_only {
+            display: none;
```
Comment (aab-odoo) [inline addons/web/static/src/scss/form_view.scss]: this is probably not necessary as there is already the other rule

File: addons/web/static/src/js/views/list/list_renderer.js [hunk 2]
```diff
@@ -749,6 +749,9 @@ var ListRenderer = BasicRenderer.extend({
             return $th;
         }
         var description = node.attrs.string || field.string;
+        if (node.attrs.class && node.attrs.class.indexOf('oe_read_only') !== -1) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/list/list_renderer.js]: handle oe_edit_only has well then...

File: addons/web/static/tests/views/form_tests.js [hunk 3]
```diff
@@ -4559,6 +4559,45 @@ QUnit.module('Views', {
         form.destroy();
     });
 
+    QUnit.test('field properly hide in edit mode in list view', async function (assert) {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form_tests.js]: oe_read_only className is handled in list views

File: addons/web/static/tests/views/form_tests.js [hunk 4]
```diff
@@ -4559,6 +4559,45 @@ QUnit.module('Views', {
...
+        var form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form string="Partners">' +
+                    '<sheet>' +
+                        '<field name="p">' +
+                            '<tree editable="top">' +
+                                '<field name="foo"/>' +
+                                '<field name="display_name" class="text-left oe_read_only" string=" "/>' +
```
Comment (aab-odoo) [inline addons/web/static/tests/views/form_tests.js]: keep it as simple as possible: why text-left, why string=" "?

File: addons/web/static/src/js/views/list/list_renderer.js (hunk 2 above)
Comment (aab-odoo) [reply addons/web/static/src/js/views/list/list_renderer.js]: maybe do it above, so that it also works for non-field th (line 748)

**Comment by msh-odoo** [comment]
@aab-odoo changes has been done, kept separate test for ```oe_read_only``` and ```oe_edit_only``` for simplicity, if you want then I can change test and test both thing 1 test

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**